enhancements/windows-containers/health-management.md

```diff
@@ -0,0 +1,311 @@
+The `services` key contains all data required to configure the required Windows services on any instance that is to be
+added to the cluster as a Node. The proposed schema is as follows:
+```json
+[
+  {
+    "name": "name of the Windows service",
+    "command": "the command that will be ran",
+    "dependencies": [
+      "name of a service that this service is dependent on"
+    ],
+    "nodeVariablesToSubstitute": {
+      "name": "string within command field that will be substituted",
+      "jsonPathNodeObject": "The jsonPath of a field within the instance's Node object"
+    },
+    "powershellVariablesToSubstitute": {
+      "name": "string within command field that will be substituted",
+      "path": "location of the PowerShell script to be ran, in order to get it's output"
+    },
+    "priority": "integer that will be used to order the creation of the services, priority 1 is created first"
+  }
+]
+```
+
+The nodeVariablesToSubsitute and powershellVariablesToSubstitute fields will be used by WICD when processing the
+ConfigMap. They are needed as some Windows services have arguments that require values sourced from either the
+Node object associated with the instance, or data from the instance itself. These fields allow for a ConfigMap that is
+generic enough to apply to all Windows instances being configured, by providing WICD instructions for how to retrieve
+certain values.
+
+For example, if the service ConfigMap had an entry with the following data:
+```
+{
+  "name": "new-service",
+  "command": "C:\new-service --variable-arg1=NODE_NAME --variable-arg2=NETWORK_IP",
+  "nodeVariablesToSubstitute": {
+    "name": "NODE_NAME",
+    "jsonPathNodeObject": "metadata.name"
+  },
+  "powershellVariablesToSubstitute": {
+    "name": "NETWORK_IP",
+    "path": "C:\k\scripts\get_net_ip.ps"
+  },
+  "dependencies": [],
+  "priority": 2
+}
+```
+
+WICD would know to create the service named `new-service` with the value of the --variable-arg1 argument set to the
+name of the instance's Node, and the output of the PowerShell script located at `C:\k\scripts\get_net_ip.ps`, as the
+value of the argument --variable-arg2. WICD will be aware of the proper Node object, as it searching for a Node with an
+internal IP equivalent to the IP of the instance.
+This service has no dependencies, and should be created only after all services with priority 1 have been created.
+
+The `files` key contains the path and checksum of files copied to the instance by WMCO. This will be used to validate
+that essential files have not been tampered with.
+```json
+{
+  "path": "filepath",
+  "checksum": "checksum of file, generated at compile time"
+}
+```
+
+### Node configuration: WMCO responsibilities
+
+1) For a given Machine or BYOH instance, WMCO copies the payload to the instance.
+2) WMCO runs WICD with the bootstrap command, this results in the node object being created.
+3) WMCO annotates the Node object with the `desiredVersion` annotation.
```

Comment:
Please explain why this `desiredVersion` is required. 

How do you handle a user stomping on the `desiredVersion`?